qutip/core/options.py

```diff
@@ -1,9 +1,50 @@
-from ..optionsclass import optionsclass
+from ..settings import settings
 
 __all__ = ["CoreOptions"]
 
-@optionsclass("core")
-class CoreOptions:
+try:
+    import cython
+    _use_cython = True
+except ImportError:
+    _use_cython = False
+
+
+class QutipOptions:
+    """
+    Class for basic functionallity for qutip's options.
+
+    Define basic method to wrap an ``options`` dict.
+    Default options are in a class _options dict.
+    """
+    _options = {}
+
+    def __init__(self, **options):
+        self.options = self._options.copy()
+        for key in set(options) & set(self.options):
+            self[key] = options.pop(key)
+        if options:
+            raise KeyError(f"Options {set(options)} are not supported.")
+
+    def __contains__(self, key):
+        return key in self.options
+
+    def __getitem__(self, key):
+        # Let the dict catch the KeyError
+        return self.options[key]
+
+    def __setitem__(self, key, value):
+        # Let the dict catch the KeyError
+        self.options[key] = value
+
+    def __repr__(self):
+        out = [f"<{self.__class__.__name__}("]
+        for key, value in self.options.items():
+            out += [f"    '{key}': {repr(value)},"]
+        out += [")>"]
+        return "\n".join(out)
+
+
+class CoreOptions(QutipOptions):
     """
     Settings used by the Qobj.  Values can be changed in qutip.settings.core.
```

Comment:
Is `Settings used by the Qobj` correct? It feels like these are options for QuTiP as a whole.